src/prefect/tasks/redis/redis_tasks.py

```diff
@@ -0,0 +1,214 @@
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+import redis
+
+
+class RedisSet(Task):
+    """
+    Task for setting a Redis key-value pair.
+
+    Args:
+        - connection (Redis Connection , optional): if not provided, connection will be established using
+            host, port, db, and password parameters
+        - host (str, optional): name of Redis host, defaults to 'localhost'
+        - port (int, optional): Redis port, defaults to 6379
+        - db (int, optional): redis database index, defaults to 0
+        - password (str, optional): Redis password, defaults to None
```

Comment:
We might want to use a Prefect Secret here for added security.  So instead of a password as the argument, it would change to `password_secret` which is a string referring to the name of a Prefect Secret which stores the password.